### PR TITLE
Turn on quota related admission control plug-ins

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Turn on the admission control plug-ins from upstream Kubernetes that control quota usage.

This change will cause LimitRange and ResourceQuota to be enforced on admission.

Note: If and only if you add a ResourceQuota to a project, all create and update operations in that project will not work until #1047 is merged.  This is because the manager needs to update actual usage stats before new requests are accepted by these plug-ins.

Review please @jwforres @smarterclayton @deads2k

### CODE REVIEW

**Comment by smarterclayton** [comment]
Do we want this enabled by default?

**Comment by derekwaynecarr** [comment]
Yes - it's a no-op when ResourceQuota and LimitRange are not used, so there is no reason to not have them on.

**Comment by derekwaynecarr** [comment]
Well, let me rephrase, it will be a no-op after my change to upstream makes these two plug-ins cache backed.  But I am waiting for https://github.com/GoogleCloudPlatform/kubernetes/pull/4453 to get merged to send the follow-on PR.

**Comment by derekwaynecarr** [comment]
The upstream change to make the admission control plug-in cache based is here:
https://github.com/GoogleCloudPlatform/kubernetes/pull/4749

Once we rebase on a version with that change, we can enable the plug-in by default.

**Comment by smarterclayton** [comment]
Rebase

**Comment by derekwaynecarr** [comment]
Rebased.

@deads2k @liggitt - this is easier to follow now with the new code split in start.go.

When starting Kube, I start the admission control plug-ins that we require in base Kube if we were to run on Kube.  Right now for Origin based master, we just run the AlwaysAdmit until we have our unique plug-ins that need to handle our own custom fields on LimitRange (i.e. layer size, etc.)

File: pkg/cmd/server/kube_master.go [hunk 1]
```diff
@@ -44,6 +51,11 @@ func (cfg Config) BuildKubernetesMasterConfig(requestContextMapper kapi.RequestC
 		masterIP = addrs[0]
 	}
 
+	// in-order list of plug-ins that should intercept admission decisions
+	// TODO: add NamespaceExists
+	admissionControlPluginNames := []string{"LimitRanger", "ResourceQuota", "AlwaysAdmit"}
```
Comment (liggitt) [inline pkg/cmd/server/kube_master.go]: Is AlwaysAdmit needed at the end of the chain? I thought admission plugins only gave errors, so if it made it to the end of the chain without an error, it was implicitly accepted

Comment (derekwaynecarr) [inline pkg/cmd/server/kube_master.go]: You are correct.

Updating.

**Comment by liggitt** [comment]
LGTM

**Comment by smarterclayton** [comment]
[merge]
